fsspec/implementations/hdfs.py

```diff
@@ -1,17 +1,93 @@
+import errno
+import os
+import re
+import secrets
+import shutil
+import subprocess
 import weakref
+from contextlib import suppress
+from functools import partial, wraps
 
 from pyarrow.hdfs import HadoopFileSystem
 
-from ..spec import AbstractFileSystem
-from ..utils import infer_storage_options
+from fsspec.spec import AbstractFileSystem
+from fsspec.utils import infer_storage_options
 
+_NOT_FOUND = os.strerror(errno.ENOENT)
 
+
+def wrap_exceptions(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except OSError as exception:
```

Comment:
actually, this might no longer be needed, https://issues.apache.org/jira/browse/ARROW-6248 was fixes in april and looks like the fix was included in the already released versions.